samples/bluetooth/tmap_bmr/src/bap_broadcast_sink.c

```diff
@@ -0,0 +1,414 @@
+struct bt_bap_stream *streams_p[ARRAY_SIZE(streams)];
+
+static struct bt_audio_codec_cap codec = BT_AUDIO_CODEC_LC3_CONFIG_48_2(
+	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+
+/* Create a mask for the maximum BIS we can sync to using the number of streams
+ * we have. We add an additional 1 since the bis indexes start from 1 and not
+ * 0.
+ */
+static const uint32_t bis_index_mask = BIT_MASK(ARRAY_SIZE(streams) + 1U);
+static uint32_t bis_index_bitfield;
+
+
+static void stream_started_cb(struct bt_bap_stream *stream)
+{
+	printk("Stream %p started\n", stream);
+}
+
+static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
+{
+	printk("Stream %p stopped with reason 0x%02X\n", stream, reason);
+}
+
+static void stream_recv_cb(struct bt_bap_stream *stream,
+			   const struct bt_iso_recv_info *info,
+			   struct net_buf *buf)
+{
+	static uint32_t recv_cnt;
+
+	recv_cnt++;
+	if ((recv_cnt % 20U) == 0U) {
+		printk("Received %u total ISO packets\n", recv_cnt);
+	}
+}
+
+static struct bt_bap_stream_ops stream_ops = {
+	.started = stream_started_cb,
+	.stopped = stream_stopped_cb,
+	.recv = stream_recv_cb
+};
+
+static struct bt_pacs_cap cap = {
+	.codec_cap = &codec,
+};
+
+static uint16_t interval_to_sync_timeout(uint16_t interval)
+{
+	uint32_t interval_ms;
+	uint16_t timeout;
+
+	/* Ensure that the following calculation does not overflow silently */
+	__ASSERT(SYNC_RETRY_COUNT < 10, "SYNC_RETRY_COUNT shall be less than 10");
+
+	/* Add retries and convert to unit in 10's of ms */
+	interval_ms = BT_GAP_PER_ADV_INTERVAL_TO_MS(interval);
+	timeout = (interval_ms * SYNC_RETRY_COUNT) / 10;
+
+	/* Enforce restraints */
+	timeout = CLAMP(timeout, BT_GAP_PER_ADV_MIN_TIMEOUT,
+			BT_GAP_PER_ADV_MAX_TIMEOUT);
+
+	return timeout;
+}
+
+static void sync_broadcast_pa(const struct bt_le_scan_recv_info *info,
+			      uint32_t broadcast_id)
+{
+	struct bt_le_per_adv_sync_param param;
+	int err;
+
+	/* Unregister the callbacks to prevent broadcast_scan_recv to be called again */
+	bt_le_scan_cb_unregister(&broadcast_scan_cb);
+	err = bt_le_scan_stop();
+	if (err != 0) {
+		printk("Could not stop scan: %d", err);
+	}
+
+	bt_addr_le_copy(&param.addr, info->addr);
+	param.options = 0;
+	param.sid = info->sid;
+	param.skip = PA_SYNC_SKIP;
+	param.timeout = interval_to_sync_timeout(info->interval);
+	err = bt_le_per_adv_sync_create(&param, &bcast_pa_sync);
+	if (err != 0) {
+		printk("Could not sync to PA: %d", err);
+
+	} else {
+		bcast_id = broadcast_id;
+	}
+}
+
+static bool scan_check_and_sync_broadcast(struct bt_data *data, void *user_data)
+{
+	uint32_t *broadcast_id = user_data;
+	struct bt_uuid_16 adv_uuid;
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return true;
+	}
+
+	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+		return true;
+	}
+
+	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
+		if (!bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_TMAS)) {
```

Comment:
Here you are comparing `adv_uuid.uuid` to two different UUIDs; that doesn't really make any sense